### PR TITLE
#11170: Marking a older as ignored. Handling missing cases for Catego…

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
…ry and Hidden fields. Adjusting junit tests. Fixing small error in JSON. Adding Javadoc.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/structure/factories/FieldFactory.java [hunk 1]
```diff
@@ -97,22 +136,28 @@ public static Field getFieldByStructure(String structureInode, String fieldName)
...
+	 * Saves a field in a Content Type.
+	 * 
+	 * @param oldField
+	 *            - The legacy Field object.
+	 * @return The Field object that was saved.
+	 * @throws DotHibernateException
+	 *             An error occurred when saving the field.
+	 */
+	public static Field saveField(Field oldField) throws DotHibernateException {
+		if (Field.FieldType.HOST_OR_FOLDER.toString().equals(oldField.getFieldType())
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/structure/factories/FieldFactory.java]: why do not use a set here instead of a big If

Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/portlets/structure/factories/FieldFactory.java]: And @jcastro-dotcms is this the right place for this code?  I'm not looking at it but shouldn't it be handled lower down in the new API?

Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/portlets/structure/factories/FieldFactory.java]: or here?  

https://github.com/dotCMS/core/blob/master-4.1/dotCMS/src/main/java/com/dotcms/contenttype/transform/field/LegacyFieldTransformer.java#L102

Comment (erickgonzalez) [inline dotCMS/src/main/java/com/dotmarketing/portlets/structure/factories/FieldFactory.java]: It seems that in the new FieldFactory this is already checked https://github.com/dotCMS/core/blob/master-4.1/dotCMS/src/main/java/com/dotcms/contenttype/business/FieldFactoryImpl.java#L230

**Review by jgambarios — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**